doc/user/content/guides/temporal-filters.md

```diff
@@ -202,3 +206,57 @@ SELECT *, mz_logical_timestamp() FROM valid;
 ```
 
 If you run this query again after 1.67 minutes from the first insertion, you'll see only two results, because the first result no longer satisfies the predicate.
+
+### Bucketing
+
+It is common to use buckets to represent the time window where a row belongs. Materialize [date functions](/sql/functions/#date-and-time-func) will help us here. <br/>
+Let's continue with the last example:
+
+```sql
+CREATE MATERIALIZED VIEW valid_buckets AS
+SELECT
+  content,
+  insert_ms,
+  delete_ms,
+  -- Trunc timestamps to the minute and use them as buckets
+  DATE_TRUNC('minute', to_timestamp(insert_ms / 1000)) as insert_bucket,
+  DATE_TRUNC('minute', to_timestamp(delete_ms / 1000)) as delete_bucket
+FROM events
+WHERE mz_logical_timestamp() >= insert_ms
+  AND mz_logical_timestamp() < delete_ms;
+```
+
+Re-insert some rows to make sure there is data available, and run the following query to see all the records and their respective buckets:
+
+```sql
+SELECT * FROM valid_buckets;
+```
+
+```nofmt
+ content |   insert_ms   |   delete_ms   |     insert_bucket      |     delete_bucket
+---------+---------------+---------------+------------------------+------------------------
+ goodbye | 1647551752960 | 1647551952960 | 2022-03-17 21:15:00+00 | 2022-03-17 21:19:00+00
+ hello   | 1647551752687 | 1647551852687 | 2022-03-17 21:15:00+00 | 2022-03-17 21:17:00+00
+ welcome | 1647551752694 | 1647551902694 | 2022-03-17 21:15:00+00 | 2022-03-17 21:18:00+00
+(3 rows)
+```
+
+It's is possible to go even more beyond aggregating by the buckets:
```

Comment:
There's kind of a fine line here, since the most common usage of `date_trunc` and `date_bin` would be directly with a `GROUP BY` and unrelated to temporal filters. This makes me wonder if we should make the guide more generic to windowed aggregations instead of trying to jam these functions into a guide that is so far specific to temporal filters.

Reply:
Let's do this as a next step!